pkg/reconciler/timeout_handler_test.go

```diff
@@ -259,3 +259,80 @@ func TestWithNoFunc(t *testing.T) {
 	testHandler.CheckTimeouts()
 
 }
+
+// TestGetTaskRunEndTime checks that the end time calculated for a given taskrun is consistent
+// and correctly uses the default taskrun timeout.
+func TestGetTaskRunEndTime(t *testing.T) {
+	timeNow := time.Now()
+	timeInTwoSeconds := timeNow.Add(2 * time.Second)
+	timeAfterDefaultTimeout := timeNow.Add(defaultTimeout)
+	testCases := []struct {
+		startTime       time.Time
+		optionalTimeout time.Duration
+		expectedEndTime time.Time
+	}{
+		{
+			startTime:       timeNow,
```

Comment:
style nit: if you add a description to each case as well, you can pass that to t.Run like I explained below. Then the test cases are a bit easier to read and failures are easier to debug.